tests/simd/hamming_distance.mlir

```diff
@@ -1,5 +1,5 @@
 // RUN: heir-opt --secretize=entry-function=hamming --wrap-generic --canonicalize --cse \
-// RUN:   --full-loop-unroll --insert-rotate --cse --canonicalize \
+// RUN:   --full-loop-unroll --cse --canonicalize --insert-rotate --cse --canonicalize \
```

Comment:
I think it would make sense to define this as a pass pipeline, so that we can more easily ensure all `simd` tests use the same pipeline (it's easy to accidentally build a set of passes where, for each program, there's a working pipeline, but there's no single pipeline that works for every program - don't ask me how I now 😉)

Reply:
I have been playing around with `secret-to-bgv` so I will probbaly start adding these passes to a pipeline like that!

Reply:
Leaving it for later.